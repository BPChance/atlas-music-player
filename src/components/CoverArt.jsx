import placeholder from "../assets/placeholder.svg";

export default function CoverArt() {
  return (
    <div className="w-full max-w-[400px] aspect-square overflow-hidden rounded-lg shadow-2xl">
  <img
    src={placeholder}
    alt="placeholder"
    className="h-full w-full object-cover"
  />
</div>

  );
}
