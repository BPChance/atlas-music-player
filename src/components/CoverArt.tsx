type CoverArtProps = {
  coverArtUrl: string;
};

export default function CoverArt({ coverArtUrl }: CoverArtProps) {
  return (
    <div className="aspect-square w-full max-w-[400px] overflow-hidden rounded-lg shadow-2xl">
      <img
        src={coverArtUrl}
        alt="placeholder"
        className="h-full w-full object-cover"
      />
    </div>
  );
}
