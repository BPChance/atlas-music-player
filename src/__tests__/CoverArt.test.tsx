import { render } from "@testing-library/react";
import CoverArt from "../components/CoverArt";
import { describe, expect, it } from "vitest";

describe("CoverArt", () => {
  it("renders a coverarturl", () => {
    const { container } = render(<CoverArt coverArtUrl="example" />);
    expect(container).toMatchSnapshot();
  });

  it("renders nothing when there is no coverarturl", () => {
    const { container } = render(<CoverArt coverArtUrl="" />);
    expect(container).toMatchSnapshot();
  });
});
