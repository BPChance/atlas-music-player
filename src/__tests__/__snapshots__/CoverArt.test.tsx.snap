// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt > renders a coverarturl 1`] = `
<div>
  <div
    class="aspect-square w-full max-w-[400px] overflow-hidden rounded-lg shadow-2xl"
  >
    <img
      alt="placeholder"
      class="h-full w-full object-cover"
      src="example"
    />
  </div>
</div>
`;

exports[`CoverArt > renders nothing when there is no coverarturl 1`] = `
<div>
  <div
    class="aspect-square w-full max-w-[400px] overflow-hidden rounded-lg shadow-2xl"
  >
    <img
      alt="placeholder"
      class="h-full w-full object-cover"
    />
  </div>
</div>
`;
